src/iop/hazeremoval.c

```diff
@@ -655,11 +654,11 @@ static void guided_filter_tiling(const_rgb_image imgg, gray_image img, gray_imag
   box_mean(a_g, a_g, w);
   box_mean(a_b, a_b, w);
   box_mean(b, b, w);
-  for(int j_imgg = source.lower; j_imgg < source.upper; j_imgg++)
+  for(int j_imgg = target.lower; j_imgg < target.upper; j_imgg++)
   {
-    size_t k = (size_t)(j_imgg - source.lower) * width;
-    size_t l = source.left + (size_t)j_imgg * imgg.width;
-    for(int i_imgg = source.left; i_imgg < source.right; i_imgg++, k++, l++)
+    size_t k = (size_t)(j_imgg - source.lower) * width + (target.left - source.left);
+    size_t l = source.left + (size_t)j_imgg * imgg.width + (target.left - source.left);
```

Comment:
while this looks all very plausible to me, why do both the source and the target coordinate k and l get offset by the same value? because the buffer layouts are the same and source and target just give the region of interest to process?

Reply:
I reformulated the code slightly and added some comments to clarify the rather tricky index arithmetic.